google/cloud/spanner/samples/samples.cc

```diff
@@ -86,19 +86,56 @@ void CreateInstance(google::cloud::spanner::InstanceAdminClient client,
                   .Build())
           .get();
   if (!instance) throw std::runtime_error(instance.status().message());
-  std::cout << "Created instance [" << in << "]\n";
+  std::cout << "Created instance [" << in << "]:\n" << instance->DebugString();
 }
 //! [END spanner_create_instance] [create-instance]
 
+// [START spanner_create_instance_with_processing_units]
+void CreateInstanceWithProcessingUnits(
+    google::cloud::spanner::InstanceAdminClient client,
+    std::string const& project_id, std::string const& instance_id,
+    std::string const& display_name, std::string const& config) {
+  namespace spanner = google::cloud::spanner;
+  spanner::Instance in(project_id, instance_id);
+
+  std::string instance_config =
+      "projects/" + project_id + "/instanceConfigs/" + config;
+  auto instance =
+      client
+          .CreateInstance(
+              spanner::CreateInstanceRequestBuilder(in, instance_config)
+                  .SetDisplayName(display_name)
+                  .SetProcessingUnits(500)
+                  .SetLabels({{"cloud_spanner_samples", "true"}})
+                  .Build())
+          .get();
+  if (!instance) throw std::runtime_error(instance.status().message());
+  std::cout << "Created instance [" << in << "]:\n" << instance->DebugString();
+
+  instance = client.GetInstance(in);
+  if (!instance) throw std::runtime_error(instance.status().message());
+  std::cout << "Instance " << in << " has " << instance->processing_units()
```

Comment:
Does the example require this call to `.GetInstance()`?  That seems slightly odd, as the `.CreateInstance()` operation should return processing units too.  If it is not needed, may elide it to keep the example simpler.  If it is need it maybe a comment explaining why?

Reply:
I have a recollection of asking that very question, but I cannot find a trace of the answer.  So, I've asked again and will report back here.

If I was pressed to dig up a memory, I think it would be that the `CreateInstance()` actually didn't return the processing units, whether for good reason or bad.

Reply:
Feel free to merge as-is and cleanup when you get an answer.  I did not mean to block you.

Reply:
The response was, _"the call for getInstance() was to essentially have a test for the API surfaces that were touched by the published protos.  Also to show how to use the field mask."_

The "field mask" stuff is hidden from C++ users (at least while we have a non-generated admin API).  But otherwise we don't have anything showing `GetInstance()` returning `processing_units`, so I'm inclined to leave it because of that.  And also to maintain some similarity with the sample in other languages.

Reply:
SGTM.